### ISSUE TITLE
Add .andThen(Runnable onSuccess, Runnable onFailure) to await API

### ISSUE THREAD

**Issue by slawekradzyminski** [FluentLenium/FluentLenium#940]
I'd like to write

```java
await()
    .atMost(Duration.ofSeconds(2))
    .untilCondition(() -> getDriver().getWindowHandles().size() > 1)
    .andThen(
        () -> { /* logic for case with new window here */ },
        () -> { /* logic for case without new window here */ }
    );
```

**Comment by picimako** [comment]
Hi @slawekradzyminski 

I did some investigation:
- In order to run the `andThen()` method properly, `until(Supplier)` should save the result of the assertion and not let it throw an exception if there were any during the wait.
- However that would be problematic when there is no `andThen()` call after `until()` because then the wait/assertion wouldn't fail, it would just save the result. And cannot know in advance whether `andThen()` is going to be called. (Assuming that that the `untilCondition()` method refers to the already available `until(Supplier)` one and not being introduced for the `andThen()` method.)

An alternate solution to overcome this might be to have separate `onFailure()` and `onSuccess()` (or similarly named) methods called before `until()` (similarly to how RxJava Observables can be configured).

```java
await()
    .onFailure(() -> { /* logic for case without new window here */ })
    .onSuccess(() -> { /* logic for case with new window here */ })
    .atMost(Duration.ofSeconds(2))
    .untilCondition(() -> getDriver().getWindowHandles().size() > 1);
```

Pros:
- it would make it easier to configure them separately, or just one of them if that's desired by users
- this could be configured properly with probably any of the until methods

Cons:
- it ruins a little bit the fluent nature of the await() chained calls.

To improve the situation in Cons I could imagine an overloaded version of `await()` parameterized with the `onFailure()` and `onSuccess()` actions:

```java
await(
    () -> { /* logic for case with new window here */ },
    () -> { /* logic for case without new window here */ })
    .atMost(Duration.ofSeconds(2))
    .until(() -> getDriver().getWindowHandles().size() > 1);
```

basically hiding the following calls:
```java
public FluentWait await(Runnable onSuccess, Runnable onFailure) {
    return await().onSuccess(onSuccess).onFailure(onFailure);
}
```

Potentially affected classes:
- `FluentWebElement`
- `FluentList`
- any class/interface that implements `org.fluentlenium.core.wait.AwaitControl`

**Comment by slawekradzyminski** [comment]
Hi @picimako thanks for investigation :)

Overloaded method seems OK for me, have you tried to implement it?

**Comment by picimako** [comment]
@slawekradzyminski 
Not yet.

**Comment by slawekradzyminski** [comment]
The goal is to provide functionality. Next release will be with constantly delayed Selenium 4 so there is still some time to squeeze it :)
